### PR TITLE
Migrate to async listener

### PR DESCRIPTION
**Pull request by starkillerOG**
Demo script:

```
import asyncio
from xiaomi_gateway import AsyncXiaomiGatewayMulticast

gateway_ip = "192.168.1.IP"

async def asyncio_demo(loop):
    def callback_func_gateway(data):
        print("Gateway callback")
        print(data)

    
    multicast = AsyncXiaomiGatewayMulticast(interface = "any")

    await multicast.start_listen()

    multicast.register_gateway(gateway_ip, callback_func_gateway)

    await asyncio.sleep(30)

    multicast.stop_listen()


if __name__ == "__main__":
    loop = asyncio.new_event_loop()
    loop.run_until_complete(asyncio_demo(loop))
```

### CODE REVIEW

**Comment by starkillerOG** [comment]
@syssi could you have a look and merge?

File: xiaomi_gateway/__init__.py [hunk 1]
```diff
@@ -19,50 +20,186 @@
...
+    def Register_gateway(self, ip, callback):
+        """Register a Gateway to this Multicast listener."""
+        if ip in self._registered_callbacks:
+            _LOGGER.error(
+                "A callback for ip '%s' was already registed, overwriting previous callback",
+                ip,
+            )
+        self._registered_callbacks[ip] = callback
+
+    def Unregister_gateway(self, ip):
```
Comment (syssi) [inline xiaomi_gateway/__init__.py:109]: Please use snake case for method names.

**Comment by syssi** [comment]
I will test the new implementation later the week and provide some additional feedback! Thanks for your time!

**Comment by starkillerOG** [comment]
@syssi since the new async listener will require some small changes in HomeAssistant, you can't yet test with HomeAssistant.
I posted my demo script in the discription of this PR, if you push buttons/open doors (with aqara) while running the script, you schould see the events coming in. (just change the IP to the IP of your aqara gateway).

**Comment by starkillerOG** [comment]
@syssi could you review/merge this?

**Comment by starkillerOG** [comment]
@syssi gentile reminder.

**Comment by ReWiG** [comment]
I would like to know what is the status of the verification of this PR? Is additional help needed?

**Comment by syssi** [comment]
@ReWiG If you could do a review and some testing I would be happy. I would like to make sure this change improves the situation and doesn't make it worse.

**Comment by starkillerOG** [comment]
@ReWiG you can test this using the demo script in the discription.
@syssi Note that I did test it myself using 2 gateways, but I do get you want to have it tested on a diffrent network by someone else.

**Comment by starkillerOG** [comment]
@syssi can you just merge this and release a new version?
We can always easily revert to a previous version....

**Comment by starkillerOG** [comment]
@Danielhiversen would you be so kind to also release a new version to pypi?

**Comment by Danielhiversen** [comment]
Done

**Comment by Danielhiversen** [comment]
Ahh, the .travis builder is not working anymore.
Will look into it.

**Comment by starkillerOG** [comment]
@Danielhiversen thank you very much for releasing the new version!
The HomeAssistant PR is here: https://github.com/home-assistant/core/pull/74979

**Pull request merged**